File: Doctor Appointment Portal/DoctorAppointmentPortalClassLibrary/Models/PatientDetailsModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DoctorAppointmentPortalClassLibrary.Models
{
    public class PatientDetailsModel
    {
        public int PatientId { get; set; } 
        public int DoctorId { get; set; }
        public string FirstName { get; set; } 
        public string LastName { get; set; }
        public string Gender { get; set; }
        public DateTime DateOfBirth { get; set; }
        public string ContactNumber { get; set; } 
        public string Email { get; set; } 
        public string Description { get; set; } 

        public string InsuranceId { get; set; } 
        public string MedicalHistory { get; set; }
        public string MedicalConcern { get; set; }

        // Address Information
        public int StateId { get; set; }
        public int DistrictId { get; set; }
        public int TalukaId { get; set; }
        public int CityId { get; set; }
        public string AddressLine { get; set; }
        public string Pincode { get; set; }
        public int AddressId { get; set; }
        public string Address { get; set; }

        // Status & Metadata
        public bool IsActive { get; set; }
        public int CreatedBy { get; set; }
        public DateTime CreatedOn { get; set; }
        public int? ModifiedBy { get; set; }
        public DateTime? ModifiedOn { get; set; }

        // Dropdown / List data
        public List<StatesModel> StatesList { get; set; } 
        public List<DistrictsModel> DistrictsList { get; set; }
        public List<TalukasModel> TalukasList { get; set; }
        public List<CitiesModel> CitiesList { get; set; } 
    }
}

File: DoctorAppointmentApp/src/app/core/models/PatientDetailsModel.ts
import { CitiesModel } from "./CitiesModel";
import { DistrictsModel } from "./DistrictsModel";
import { DoctorAvailabilitiesModel } from "./DoctorAvailabilitiesModel";
import { DoctorUnavailabilityModel } from "./DoctorUnavailablityModel";
import { StatesModel } from "./StatesModel";
import { TalukasModel } from "./TalukasModel";

export class PatientDetailsModel {

    DoctorId: number = 0;
    FirstName?: string = '';
    LastName?: string = '';
    Gender?: string = '';
    DateOfBirth?: string = '';            
    ContactNumber?: string = '';          
    Email?: string = '';                  
    Description?: string = '';
    InsuranceId : string;
    MedicalHistory : string;
    MedicalConcern : string;

     // Address Information
     StateId?: number | null = null;
     DistrictId?: number | null = null;
     TalukaId?: number | null = null;
     CityId?: number | null = null;
     AddressLine?: string = '';
     Pincode?: string = '';
 
     IsActive: boolean = false;
     CreatedBy: number | null = null;
     AddressId: number | null = null;
     Address?: string = '';

     StatesList : StatesModel[];
     DistrictsList : DistrictsModel[];
     TalukasList : TalukasModel[];
     CitiesList : CitiesModel[];

}